manualdirectdebit.php

```diff
@@ -186,11 +186,36 @@ function manualdirectdebit_civicrm_navigationMenu(&$menu) {
  *
  */
 function manualdirectdebit_civicrm_postProcess($formName, &$form) {
+
   $action = $form->getAction();
 
-  if ($formName == 'CRM_Contribute_Form_Contribution' && $action == CRM_Core_Action::ADD) {
-    $manualDirectDebit = new CRM_ManualDirectDebit_Hook_PostProcess_Contribution_DirectDebitMandate($form);
-    $manualDirectDebit->checkPaymentOptionToCreateMandate();
+  switch ($formName){
+    case "CRM_Contribute_Form_Contribution":
+      if($action == CRM_Core_Action::ADD){
+        $manualDirectDebit = new CRM_ManualDirectDebit_Hook_PostProcess_Contribution_DirectDebitMandate($form);
+        $manualDirectDebit->checkPaymentOptionToCreateMandate();
+      }
+    break;
+
+    case "CRM_Member_Form_Membership":
+      if($action == CRM_Core_Action::ADD){
+        $paymentInstrumentId = $form->getVar('_submitValues') ['payment_instrument_id'];
+        if (CRM_ManualDirectDebit_Common_DirectDebitDataProvider::isPaymentMethodDirectDebit($paymentInstrumentId)) {
+          $manualDirectDebit = new CRM_ManualDirectDebit_Hook_PostProcess_Membership_DirectDebitMandate($form);
+          $manualDirectDebit->fetchMandateData();
```

Comment:
Yeah, here you can really see the problem with the method's name. I didn't understand why you would need to fetch the data and not do anything else when post-processing the form. Then I followed the call and saw `fetchMandateData()` really is used to save the data.